dev/com.ibm.ws.wssecurity_fat.wsscxf/fat/src/com/ibm/ws/wssecurity/fat/cxf/usernametoken/CheckpointRestoreTimeoutTests.java

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package com.ibm.ws.wssecurity.fat.cxf.usernametoken;
+
+import static componenttest.annotation.SkipForRepeat.EE9_FEATURES;
+import static componenttest.annotation.SkipForRepeat.NO_MODIFICATION;
+
+import java.io.File;
+import java.util.Set;
+
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.config.ServerConfiguration;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.annotation.AllowedFFDC;
+import componenttest.annotation.Server;
+import componenttest.annotation.SkipForRepeat;
+import componenttest.annotation.SkipIfCheckpointNotSupported;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.rules.repeater.EE8FeatureReplacementAction;
+import componenttest.topology.impl.LibertyFileManager;
+import componenttest.topology.impl.LibertyServer;
+
+@SkipForRepeat({ NO_MODIFICATION, EE9_FEATURES })
+@SkipIfCheckpointNotSupported
+@Mode(TestMode.FULL)
+@RunWith(FATRunner.class)
+public class CheckpointRestoreTimeoutTests extends SSLTestCommon {
+
+    static private final Class<?> thisClass = CheckpointRestoreTimeoutTests.class;
+
+    static final private String serverName = "com.ibm.ws.wssecurity_fat.ssl";
+    @Server(serverName)
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        String thisMethod = "setup";
+        String copyFromFile = "";
+
+        ShrinkHelper.defaultDropinApp(server, "untsslclient", "com.ibm.ws.wssecurity.fat.untsslclient", "fats.cxf.basicssl.wssec", "fats.cxf.basicssl.wssec.types");
+        ShrinkHelper.defaultDropinApp(server, "untoken", "com.ibm.ws.wssecurity.fat.untoken");
+        // Add jvm.options
+        server.copyFileToLibertyServerRoot("serversettings/jvm.options");
+        PrepInitServer serverObject = new PrepInitServer();
+        serverObject.prepareInit(server);
+        ServerConfiguration config = server.getServerConfiguration();
+        Set<String> features = config.getFeatureManager().getFeatures();
+        if (features.contains("jaxws-2.3")) {
```

Comment:
I'm not sure I follow this logic.  This test is always going to start with a constant list of features enabled, correct?  If so will we always know if jaxws-2.3 is used already?  Why not just start with the correct server.xml for the test server we are using?